Nodejs/Product/Nodejs/Project/NodejsGeneralPropertyPage.cs

```diff
@@ -49,16 +49,45 @@ internal override CommonProjectNode Project
 
         protected override void Apply()
         {
-            this.Project.SetProjectProperty(NodeProjectProperty.NodeExePath, this.control.NodeExePath);
-            this.Project.SetProjectProperty(NodeProjectProperty.NodeExeArguments, this.control.NodeExeArguments);
             this.Project.SetProjectProperty(CommonConstants.StartupFile, this.control.ScriptFile);
             this.Project.SetProjectProperty(NodeProjectProperty.ScriptArguments, this.control.ScriptArguments);
             this.Project.SetProjectProperty(NodeProjectProperty.NodejsPort, this.control.NodejsPort);
             this.Project.SetProjectProperty(NodeProjectProperty.StartWebBrowser, this.control.StartWebBrowser.ToString());
             this.Project.SetProjectProperty(CommonConstants.WorkingDirectory, this.control.WorkingDirectory);
             this.Project.SetProjectProperty(NodeProjectProperty.LaunchUrl, this.control.LaunchUrl);
-            this.Project.SetProjectProperty(NodeProjectProperty.DebuggerPort, this.control.DebuggerPort);
             this.Project.SetProjectProperty(NodeProjectProperty.Environment, this.control.Environment);
+
+            if (this.Project.SetProjectProperty(NodeProjectProperty.SaveNodeJsSettingsInProjectFile, this.control.StoreNodeSettingsInProject.ToString()))
+            {
+                // if this setting changed we should move the properties to the right file
+                if (this.control.StoreNodeSettingsInProject)
+                {
+                    this.Project.MovePropertyToProjectFile(NodeProjectProperty.NodeExePath);
```

Comment:
>this.Project.MovePropertyToProjectFile(NodeProjectProperty.NodeExePath); [](start = 20, length = 72)

I think it would be helpful to add a comment about why we're not just removing the old value (since we're going to set the new value below).